src/logger/__tests__/MetricsLogger.test.ts

```diff
@@ -297,6 +297,36 @@ test('context is preserved across flush() calls', async () => {
   }
 });
 
+test('putMetricWithDimensions example', async () => {
+  // this metric will use the default namespace and dimensions
+  logger.putMetricWithDimensions({
+    metrics: { "MyMetric": 100 }
+  });
+
+  // this metric will use the default dimensions
+  logger.putMetricWithDimensions({
+    metrics: { "MyMetric": 100 },
+    namespace: "My Namespace"
+  });
+
+  // publish multiple metrics along multiple dimensions
+  const client = 'client';
+  const pageType = 'pageType';
+  logger.putMetricWithDimensions({
+    metrics: { 
+      Metric1: 100,
+      Metric2: 200,
+    },
+    namespace: "My Namespace",
+    dimensions: [
```

Comment:
In the implementation, the default dimensions would also be added to each of the dimension sets, right?

Reply:
That's one of the decisions we need to make. We have 2 options that I can see:
1. Automatically merge the default dimensions into the provided dimensions. In this option we need some way to allow users to override or specify that they do not want the defaults. So you get something like the following with a way to opt-out.

```js
logger.putMetricWithDimensions({
    /* ... */
    dimensions: [{ client }],
    includeDefaultDimensions: false
  });
```

2. Do not automatically merge dimensions and provide a way to fetch the default dimensions:

```js
logger.putMetricWithDimensions({
    /* ... */
    dimensions: [{ ...logger.getDefaultDimensions(), client }]
});
```

Right now, I'm leaning towards option 1 since the default behavior is closer to the rest of the library. Option 2 requires an explicit step to merge in the defaults. Functionally, you can achieve the same results with both.

Reply:
Looks good to me. 👍